ports/servoshell/app.rs

```diff
@@ -436,7 +436,10 @@ impl App {
 
         // Catch some keyboard events, and push the rest onto the WebViewManager event queue.
         webviews.handle_window_events(embedder_events);
-        if pref!(dom.gamepad.enabled) && webviews.webview_id().is_some() {
+
+        // If the Gamepad API is enabled, handle gamepad events from GilRs.
+        // Checking for current_url_string should ensure we'll have a valid browsing context.
+        if pref!(dom.gamepad.enabled) && webviews.current_url_string().is_some() {
```

Comment:
Could you please try to remove the current_url_string condition and put the handling of gamepad events at https://github.com/servo/servo/blob/5144e337acbbff246ba4dff62e1f72d212bdd844/ports/servoshell/app.rs#L456

I think what you need is just to first have servo handle the incoming window events.